src/typechecker.erl

```diff
@@ -184,10 +182,23 @@ compatible(Ty1, Ty2, Env) ->
 
 %% The first argument is a "compatible subtype" of the second.
 
--spec subtype(type(), type(), env()) -> compatible().
+-spec subtype(type(), type(), env()) -> boolean().
 subtype(Ty1, Ty2, Env) ->
+    case subtype_with_constraints(Ty1, Ty2, Env) of
+        {true, _Cs} ->
+            %% There shouldn't be any constraint really.
+            %% We could assert that the constraints are empty but this could crash
+            %% Gradualizer even for end users if there was some error in our code,
+            %% it's probably safer not to assert and just drop the constraints
+            true;
```

Comment:
Debug logging in verbose mode, would that be useful?

Reply:
Hmm, good idea! I'll try that when I get to some other work on Gradualizer.